CMakeLists.txt

```diff
@@ -2,6 +2,7 @@ cmake_minimum_required(VERSION 3.1)
 
 project(aws-crt-jni C)
 option(BUILD_DEPS "Builds aws common runtime dependencies as part of build" ON)
+option(CRT_FIPS "Whether to build with FIPS compliance" OFF)
```

Comment:
debatable: I see you failing if FIPS=ON and this is 32bit linux, but there are still a lot of other cases where we do NOT fail (e.g. Windows, Apple, musl, armv6/7)

I'm just spooked about us *thinking* we're making FIPS builds but one day a year from now, one of the many intermediate steps for it to actually be FIPS doesn't work right during the release process

Reply:
Maybe add a check later in the file like `if (CRT_FIPS AND NOT FIPS) fail fail fail`

since we only set `FIPS=ON` when going down the aws-lc path